ginivision/src/main/java/net/gini/android/vision/review/multipage/thumbnails/ThumbnailsAdapter.java

```diff
@@ -353,23 +409,64 @@ void showActivityIndicator() {
             activityIndicator.setVisibility(View.VISIBLE);
         }
 
+        void resetImageView() {
+            if (thumbnailContainer == null) {
+                return;
+            }
+            thumbnailContainer.rotateImageView(0, false);
+            thumbnailContainer.getImageView().setImageDrawable(null);
+            hideUploadIndicators();
+        }
+
+        private void hideUploadIndicators() {
+            hideActivityIndicator();
+            hideUploadIcon();
+        }
+
         void hideActivityIndicator() {
             if (activityIndicator == null) {
                 return;
             }
             activityIndicator.setVisibility(View.INVISIBLE);
         }
 
-        void resetImageView() {
-            if (thumbnailContainer == null) {
+        void hideUploadIcon() {
+            if (uploadResultIconBackground == null
+                    || uploadResultIconForeground == null) {
                 return;
             }
-            thumbnailContainer.rotateImageView(0, false);
-            thumbnailContainer.getImageView().setImageDrawable(null);
+            uploadResultIconBackground.setVisibility(View.INVISIBLE);
+            uploadResultIconForeground.setVisibility(View.INVISIBLE);
         }
 
         boolean isDragAllowed() {
-            return mViewType == ViewType.THUMBNAIL;
+            return viewType == ViewType.THUMBNAIL;
+        }
+
+        void showUploadSuccess() {
```

Comment:
Could you combine this two methods into a switch? I'm fine with this but I think it is the perfect place to have a switch for the `UploadState`

Reply:
Well they're invoked from within a switch. 😁 This ViewHolder doesn't know about the UploadState. These two methods allow the showing of upload success/failure. Under which circumstances they're invoked is imho irrelevant for the ViewHolder.